glean-core/rlb/src/lib.rs

```diff
@@ -541,5 +576,94 @@ pub fn test_reset_glean(cfg: Configuration, client_info: ClientInfoMetrics, clea
     initialize(cfg, client_info);
 }
 
+/// Sets a debug view tag.
+///
+/// When the debug view tag is set, pings are sent with a `X-Debug-ID` header with the
+/// value of the tag and are sent to the ["Ping Debug Viewer"](https://mozilla.github.io/glean/book/dev/core/internal/debug-pings.html).
+///
+/// # Arguments
+///
+/// * `tag` - A valid HTTP header value. Must match the regex: "[a-zA-Z0-9-]{1,20}".
+///
+/// # Returns
+///
+/// This will return `false` in case `tag` is not a valid tag and `true` otherwise
+/// or if the tag is set before Glean is initialized.
+pub fn set_debug_view_tag(tag: &str) -> bool {
+    if was_initialize_called() {
+        with_glean_mut(|glean| glean.set_debug_view_tag(tag))
+    } else {
+        // Glean has not been initialized yet. Cache the provided tag value.
+        if PRE_INIT_DEBUG_VIEW_TAG.get().is_none() {
+            if PRE_INIT_DEBUG_VIEW_TAG
+                .set(Mutex::new(tag.to_string()))
+                .is_err()
+            {
+                log::error!(
+                    "Unable to set the debug view tag: already set. This probably happened concurrently."
+                );
+            }
+        } else {
+            let mut lock = PRE_INIT_DEBUG_VIEW_TAG.get().unwrap().lock().unwrap();
+            *lock = tag.to_string();
+        }
+        // When setting the debug view tag before initialization,
+        // we don't validate the tag, thus this function always returns true.
+        true
+    }
+}
+
+/// Sets the log pings debug option.
+///
+/// This will return `false` in case we are unable to set the option.
+///
+/// When the log pings debug option is `true`,
+/// we log the payload of all succesfully assembled pings.
+///
+/// # Arguments
+///
+/// * `value` - The value of the log pings option
+pub fn set_log_pings(value: bool) {
+    if was_initialize_called() {
+        with_glean_mut(|glean| glean.set_log_pings(value));
+    } else {
+        PRE_INIT_LOG_PINGS.store(value, Ordering::SeqCst);
+    }
+}
+
+/// Sets source tags.
+///
+/// Ping tags will show in the destination datasets, after ingestion.
+///
+/// # Arguments
+///
+/// * `value` - A vector of at most 5 valid HTTP header values. Individual
+///   tags must match the regex: "[a-zA-Z0-9-]{1,20}".
+///
+/// # Returns
+///
+/// This will return `false` in case `value` contains invalid tags and `true`
+/// otherwise or if the tag is set before Glean is initialized.
+pub fn set_source_tags(value: Vec<String>) -> bool {
+    if was_initialize_called() {
+        with_glean_mut(|glean| glean.set_source_tags(value))
+    } else {
+        // Glean has not been initialized yet. Cache the provided source tags.
+        if PRE_INIT_SOURCE_TAGS.get().is_none() {
```

Comment:
As above the shorter `get_or_init` pattern seems better.

(I _think_ I initially came up with this `is_none` check then print if it still errors, but I'm not even sure why anymore)